packages/quantic/force-app/main/default/lwc/quanticSearchBox/quanticSearchBox.js

```diff
@@ -202,21 +202,23 @@ export default class QuanticSearchBox extends LightningElement {
   };
 
   /**
-   * Handles the selection of a suggestion.
+   * Handles the selection of a suggestion or a recent query.
    */
   selectSuggestion = (event) => {
     event.stopPropagation();
     const {value, isRecentQuery, isClearRecentQueryButton} =
       event.detail.selectedSuggestion;
+
     if (isClearRecentQueryButton) {
       this.recentQueriesList.clear();
-    } else if (isRecentQuery) {
+    }
+
+    if (isRecentQuery) {
       this.recentQueriesList.executeRecentQuery(
         this.recentQueries.indexOf(value)
       );
-    } else {
-      this.searchBox?.selectSuggestion(value);
     }
+    this.searchBox?.selectSuggestion(value);
```

Comment:
Does this send an event though? We need to make sure that if this sends an analytics event that it's the correct type, etc.

Reply:
yeah no I put the PR in draft following Mehdi's comment. What happens is that it sends 1 event, but the wrong one. It sends 2 ( execute recent queries and select suggestion) and aborts one (the recent queries)